src/ai/backend/appproxy/coordinator/config.py

```diff
@@ -287,6 +287,19 @@ class ProxyCoordinatorConfig(BaseSchema):
         ),
     ]
 
+    @property
+    def advertise_base_url(self) -> str:
+        """
+        Generate the full advertise URL with protocol.
+        Uses advertised_addr if set, otherwise falls back to bind_addr.
+        Protocol is determined by tls_advertised or tls_listen flags.
+        """
+        connection_info = (
+            self.advertised_addr if self.advertised_addr is not None else self.bind_addr
+        )
+        protocol = "https" if (self.tls_advertised or self.tls_listen) else "http"
+        return f"{protocol}://{connection_info.clean_host}:{connection_info.port}"
```

Comment:
Please add test code to match the scenarios being generated. (This would be good to set up with parametrize.)